compiler/src/dmd/errorsink.d

```diff
@@ -13,6 +13,23 @@ module dmd.errorsink;
 
 import dmd.location;
 
+/// Constants used to map compiler warnings to a specific flag.
+enum DiagnosticFlag
```

Comment:
This is a nice step forward, but is the aim to have every error have its own enum value or merely categories? The comment should reflect which one (or in between, something like a huffman code etc etc)

Reply:
Every warning. Errors shouldn't be optional.

I'd say every category - as there is more than one way to trigger some (ie: unused _things_).

Reply:
Also what is a specific flag in this case?

Reply:
> Also what is a specific flag in this case?

Something that the diagnostics reporter can use to check whether it is turned on. When passed `DiagnosticFlag.obsolete` it could check whether the obsolete param is set - or `DiagnosticFlag.override_` to check whether an override parameter is set - and if not, suppress it.

Reply:
For example, this could be a boolean array (or a bitarray to save space), in which case this DiagnosticFlag enum value would be the index into the array.

(If implementers in dmd then DiagnosticIndex might be a more accurate name).